test/test_quickbooks/test_quickbooks.py

```diff
@@ -0,0 +1,275 @@
+    mock_schedule_calendars_list_data,
+    mock_schedule_events_data,
+    mock_geolocations_data,
+)
+
+
+class TestQuickBooks(unittest.TestCase):
+    @requests_mock.Mocker()
+    def setUp(self, mock_request):
+        self.qb = QuickBooks(token="abc123")
+        self.qb.url = "https://rest.tsheets.com/api/v1/"
+
+    def tearDown(self):
+        pass
+
+    @requests_mock.Mocker()
+    def test_qb_get_request(self, mock_request, end_point=None):
+        # Arrange
+
+        # If no end_point is passed, it will default to the groups endpoint
+        if end_point is None:
+            end_point = "groups"
+
+        querystring = {"page": 1}
+        mock_request.get(requests_mock.ANY, json=mock_groups_data)
+
+        # Act
+        result = self.qb.qb_get_request(end_point=end_point, querystring=querystring)
+
+        # Assert
+        self.assertIsInstance(result, Table)
+        self.assertIsInstance(end_point, str)
+        self.assertIsInstance(querystring, dict)
+
+    @requests_mock.Mocker()
+    def test_get_groups(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_groups_data)
+
+        # Act
+        result = self.qb.get_groups()
+
+        # Assert
+        self.assertIsInstance(result, Table)
+
+    @requests_mock.Mocker()
+    def test_get_jobcodes(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_jobcodes_data)
+
+        # Act
+        result = self.qb.get_jobcodes()
+
+        # Assert
+        self.assertIsInstance(result, Table)
+
+    @requests_mock.Mocker()
+    def test_get_timesheets(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_timesheets_data)
+
+        # Act
+        result = self.qb.get_timesheets()
+
+        # Assert
+        self.assertIsInstance(result, Table)
+
+    @requests_mock.Mocker()
+    def test_get_users(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_users_data)
+
+        # Act
+        result = self.qb.get_users()
+
+        # Assert
+        self.assertIsInstance(result, Table)
+
+    @requests_mock.Mocker()
+    def test_get_schedule_calendars_list(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_schedule_calendars_list_data)
+
+        # Act
+        result = self.qb.get_schedule_calendars_list()
+
+        # Assert
+        self.assertIsInstance(result, list)
+
+    @requests_mock.Mocker()
+    def test_get_schedule_events(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_schedule_events_data)
+
+        # Act
+        result = self.qb.get_schedule_events()
+
+        # Assert
+        self.assertIsInstance(result, Table)
+
+    @requests_mock.Mocker()
+    def test_get_jobcodes_with_params(self, mock_request):
+        # Arrange
+        mock_request.get(requests_mock.ANY, json=mock_jobcodes_data)
```

Comment:
I'm a little confused why this data is here and not in `test_quickbooks_data.py`. It's not so much data as to make the tests unreadable, but it is inconsistent.